specification.md

```diff
@@ -1,15 +1,279 @@
+  maliciously.  **Impact:** Traceability in all cases.  Depending on the 
+  changes made to the SBOM, this may be detected by review of the resulting
+  SBOM.  Changes that override in-toto derived information are specifically 
+  flagged and unlikely to be accepted. XXXXXX
+* an attacker is able to compromise a SIT generator key or a SIT generator 
+  directly. **Impact:** Protection for clients who obtain the SBOMit document.
+  Traceability for other clients. Note that the SIT will contain a refernce to
+  the SBOMit document, but this also may be modified in this case.  However,
+  the client should have the in-toto layout key and so will notice this 
+  action if retrieving the SBOMit document from the the SIT.
+* an attacker is able to compromise the in-toto layout key, which serves as
+  the root-of-trust for the system.  **Impact:** Traceability.  Later analysis
+  can show this was the cause, but users will trust a new, maliciously 
+  generated layout which replaces signing keys.
+
+From (Endor Lab's top 10 OSS risks)[https://www.endorlabs.com/blog/introducing-the-top-10-open-source-software-oss-risks], 
+our design largely addresses:
+
+* OSS-RISK-1
+* OSS-RISK-2
+* OSS-RISK-5
+* OSS-RISK-9
+
+
+Orthogonal systems that can be used along with SBOMit:
+
+* A software supply chain tool which attests to the quality of steps, such 
+as FRSCA and SLSA.  This helps with knowing that an individual action is 
+actually a good security practice, could judge the quality of implementation 
+for different tools, determine if the code base metrics are indicative of 
+quality, understand if the licence is appropriate, etc.  This helps to address
+OSS-RISK-4, OSS-RISK-8, and OSS-RISK-10.  OSS-RISK-7 may also be addressed
+either by this or by end user diligence.
+
+* Some means like Sigstore's root of trust is needed to Handle how users know 
+the correct name / root of trust for the software they are installing.  This 
+relates to OSS-RISK-3.  
+
+* Recursing into components like the packages inside of a container image when 
+the build process does not otherwise do so.  This relates to OSS-RISK-6.  As 
+both tooling and in-toto adoption increase, this issue should naturally be 
+addressed as more metadata becomes available.
+
+There is also (a more detailed analysis which shows why these properties 
+hold)[TODO]
 
 ## Format/required fields
 
+### SBOMit Document
+
+#### Overview
+
+There are four items in a SBOMit document:
+
+* An in-toto layout.  This specifies the policy that the supply chain followed.
+  This includes information about the key used to sign each SIT and 
+  corresponding the SIT generation process for each.  It must be signed by the
+  layout key trusted for the SBOMit document.
+
+* A collection of in-toto metadata.  This includes sub-layouts, link metadata,
+  and attestations.  This must validate according to the in-toto layout and
+  so must be signed by the appropriate sub-layout creators and functionaries.
```

Comment:
```suggestion
  so must be signed by the appropriate creators and functionaries.
```

Reply:
the links also have creators that have to match the layout

Reply:
Aren't those functionaries?

Reply:
oh, I read this as sub-layout (creators and functionaries) rather than (sub-layout creators) and (functionaries). It's probably fine as is

Reply:
```suggestion
  so must be signed by the appropriate functionaries and sub-layout creators.
```